Userland/Libraries/LibWeb/Painting/StackingContext.cpp

```diff
@@ -503,11 +500,8 @@ Optional<HitTestResult> StackingContext::hit_test(CSSPixelPoint position, HitTes
     // 6. the child stacking contexts with stack level 0 and the positioned descendants with stack level 0.
     Optional<HitTestResult> result;
     for_each_in_subtree_of_type_within_same_stacking_context_in_reverse<PaintableBox>(paintable_box(), [&](PaintableBox const& paintable_box) {
-        // FIXME: Support more overflow variations.
-        if (paintable_box.computed_values().overflow_x() == CSS::Overflow::Hidden && paintable_box.computed_values().overflow_y() == CSS::Overflow::Hidden) {
-            if (!paintable_box.absolute_border_box_rect().contains(transformed_position.x(), transformed_position.y()))
-                return TraversalDecision::SkipChildrenAndContinue;
-        }
+        if (!can_potentionally_hit(transformed_position))
+            return TraversalDecision::SkipChildrenAndContinue;
```

Comment:
Instead of checking the set of paintables we're iterating over, we now only check the `StackingContext::paintable_box()` over and over (since that's what `can_potentially_hit` looks at). That doesn't seem right.

Reply:
Good catch, I've put the `can_potentially_hit` as a member function of `PaintableBox` so it will check the children's paintable box.